### PR TITLE
Address Code Review Comments

### PR DESCRIPTION
**Pull request by jamilfaisal**
- Remove ScaleStats and add player level to PlayerAttributes
- Add tech points to PlayerAttributes
- Move null checks to Start method of PlayerShoot
- Increase xpToNextLevel after each level up by 100
- Add 10 XP after each Mob1 is killed
- Create GameManager singleton to hold managers
- Need to set LevelManager of Mob in Mob Spawner (CooperativeDev will do this)

### CODE REVIEW

**Comment by CooperativeDev** [comment]
I see you made the following changes I asked for:

Remove ScaleStats and add player level to PlayerAttributes
Add tech points to PlayerAttributes
Move null checks to Start method of PlayerShoot
Increase xpToNextLevel after each level up by 100
Add 10 XP after each Mob1 is killed
Create GameManager singleton to hold managers
Need to set LevelManager of Mob in Mob Spawner (CooperativeDev will do this)

nice work

File: Assets/Scripts/Managers/GameManager.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+using UnityEngine;
+
+namespace Managers
+{
+    public class GameManager : MonoBehaviour
```
Comment (CooperativeDev) [inline Assets/Scripts/Managers/GameManager.cs]: Change name to MetaManager

File: Assets/Scripts/Mob/MobAbstract.cs [hunk 2]
```diff
@@ -1,9 +1,11 @@
+using Managers;
 using UnityEngine;
 
 namespace Mob
 {
     public abstract class MobAbstract : MonoBehaviour
     {
+        protected PlayerLevelManager LevelManager;
         protected int Health;
```
Comment (CooperativeDev) [inline Assets/Scripts/Mob/MobAbstract.cs]: remove unused line

File: Assets/Scripts/Player/PlayerShoot.cs [hunk 3]
```diff
@@ -8,15 +8,23 @@ public class PlayerShoot : MonoBehaviour
     {
         public GameObject bulletPrefab;
         public Transform gunEnd;
-        public PlayerAttributesManager attributesManager;
+        private PlayerAttributesManager _attributesManager;
```
Comment (CooperativeDev) [inline Assets/Scripts/Player/PlayerShoot.cs]: use MetaManager.Instance.AttributesManager

File: Assets/Scripts/Player/PlayerShoot.cs [hunk 4]
```diff
@@ -8,15 +8,23 @@ public class PlayerShoot : MonoBehaviour
...
 
         private double _fireTimer;
```
Comment (CooperativeDev) [inline Assets/Scripts/Player/PlayerShoot.cs]: change name from _fireTimer to  _shootTimer

**Review by CooperativeDev — CHANGES_REQUESTED** [review]
Better luck next time buddy!

File: Assets/Scripts/Managers/GameManager.cs (hunk 1 above)
Comment (jamilfaisal) [reply Assets/Scripts/Managers/GameManager.cs]: Done

File: Assets/Scripts/Mob/MobAbstract.cs (hunk 2 above)
Comment (jamilfaisal) [reply Assets/Scripts/Mob/MobAbstract.cs]: Done

File: Assets/Scripts/Player/PlayerShoot.cs (hunk 4 above)
Comment (jamilfaisal) [reply Assets/Scripts/Player/PlayerShoot.cs]: Done

File: Assets/Scripts/Player/PlayerShoot.cs (hunk 3 above)
Comment (jamilfaisal) [reply Assets/Scripts/Player/PlayerShoot.cs]: Done

**Review by CooperativeDev — APPROVED** [review]

**Pull request merged**